### DISCUSSION TITLE
The document contains pictures. Can you add a function to click on the picture to enlarge it?

### DISCUSSION THREAD

**Discussion by moatkon** [withastro/starlight#1220] [Q&A]
### What version of `starlight` are you using?

0.14.0

### What is your idea?

The document contains pictures. Can you add a function to click on the picture to enlarge it?

### Why is this feature necessary?

Document contains pictures

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by trueberryless** [comment]
You mean something like Lightbox: https://lokeshdhakar.com/projects/lightbox2/. Yeah, that would actually be a great feature!

**Reply by moatkon** [reply]
yes!

**Comment by stereobooster** [comment]
Alternatively you can add pan/zoom https://astro-digital-garden.stereobooster.com/recipes/svg-pan-zoom/

**Comment by delucis** [comment]
@HiDeoo also built a nice Starlight plugin for image zoom: https://github.com/HiDeoo/starlight-image-zoom

**Reply by peyloride** [reply]
How does this works with normal Astro projects? I'm not familiar with Starlight

**Comment by petrovicz** [comment]
You can use [PhotoSwipe](https://photoswipe.com/) for that.
Take a look at my [example repo](https://github.com/petrovicz/astro-photoswipe) or read my [blog post](https://dev.to/petrovicz/astro-photoswipe-549a) about adding it to an existing project.

Generally, this is for a whole image gallery, but it has an option to just open for one single image at a time.
It's a well known library with many features.

It's not as seamlessly integrated as the plugin of @delucis, but gives you more options.
You can use the built-in image optimization and it works with ViewTransitions as well.

**Comment by HiDeoo** [comment]
Considering that many possibilities can be used to cover such a feature, e.g. by many various libraries or Starlight plugins mentioned in this discussion, I don't think we have plans to implement this in Starlight core. Considering this, I'll close this discussion for now.
